src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1266,28 +1325,39 @@ string dump(BoundDagTest d)
         /// </summary>
         private class DagState
         {
+            /// <summary>
+            /// For each dag temp of a type for which we track such things (the integral types, floating-point types, and bool),
+            /// the possible values it can take on when control reaches this state.
+            /// If this dictionary is mutated after <see cref="TrueBranch"/>, <see cref="FalseBranch"/>,
+            /// and <see cref="Dag"/> are computed (for example to merge states), they must be cleared and recomputed,
+            /// as the set of possible values can affect successor states.
+            /// A <see cref="BoundDagTemp"/> absent from this dictionary means that all values of the type are possible.
+            /// </summary>
+            public ImmutableDictionary<BoundDagTemp, IValueSet> RemainingValues { get; private set; }
```

Comment:
>ImmutableDictionary [](start = 19, length = 19)

The other significant change is that we keep track, in each state of the set of possible values for each intermediate result.